tests/test_accumulator_fixed.rs

```diff
@@ -64,10 +65,25 @@ fn test_only_n_recent_values() {
     assert_eq!(a.difference(&b).count(), 0); // both set must be equal.
 }
 
-/// Compare with third-party implementation.
 #[test]
 fn test_sanity() {
-    use float_eq::assert_float_eq;
+    let mut acc = SimpleAccumulator::with_fixed_capacity(&vec![], 3);
+    acc.push(1.0);
+    acc.push(2.0);
+    assert_float_eq!(acc.mean(), 1.5, abs_all <= 1e-4);
+    acc.push(3.0);
+    assert_float_eq!(acc.mean(), 2.0, abs_all <= 1e-4);
+    acc.push(2.0);
+    assert_float_eq!(acc.mean(), 2.0, abs_all <= 1e-4);
```

Comment:
I am getting confused again:

`1, 2` -> `1.5`
push 3
`1, 2, 3` -> `2.0`
push 2
`2, 2, 3`-> `2.33` 

Shouldn't it fail? Are we no longer calculating the mean of the values that we hold and just returning the running mean? 

- If we want running mean we already had that before using watermill, `r_mean`
- Where are we calculating the mean of currently held values?

Reply:
No. This is correct. We are dropping values because we don't _want_ to keep all history to save memory. We are still computing mean and __not__ the running mean.

Reply:
@dilawar AFAIK, you mean to say that we should essentially consider mean, variance as running mean, running variance and not the mean, variance of buffer contents -- Am I right?

I just had a query that isn't a circular queue (that we had till ver `0.5.1`) more efficient than a flat queue? Or does this library `watermill` optimize everything so well that the doubling of operations (2 I/O ops during pop, maintaining and updating of both `head` and `tail` pointers) does not have a big effect?

Reply:
@purnatag Yes. I've updated my comment above. You may be right about perf but we don't have benchmarks. After this release, I'll add benches and then we can reconsider the other options. The compilers can do a lot of magic by themselves and performance by branch-prediction and caching is something some can reason about without running benchmarks.